core/storage/base_model/gae_models_test.py

```diff
@@ -209,6 +252,76 @@ def test_retrieval_of_multiple_version_models_for_fake_id(self):
             TestVersionedModel.get_multi_versions(
                 'fake_id', [1, 2, 3])
 
+    def test_commit_with_model_instance_deleted_raises_error(self):
+        model1 = TestVersionedModel(id='model_id1')
+        model1.commit(feconf.SYSTEM_COMMITTER_ID, '', [])
+        model1.delete(feconf.SYSTEM_COMMITTER_ID, 'delete')
+
+        with self.assertRaisesRegexp(
+            Exception, 'This model instance has been deleted.'):
+            model1.commit(feconf.SYSTEM_COMMITTER_ID, '', [])
+
+    def test_raises_error_with_trusted_commit_method(self):
+        model1 = TestVersionedModel(id='model_id1')
+        model1.SNAPSHOT_METADATA_CLASS = None
+        with self.assertRaisesRegexp(
+            Exception, 'No snapshot metadata class defined.'):
+            model1.commit(feconf.SYSTEM_COMMITTER_ID, '', [])
+
+        model1 = TestVersionedModel(id='model_id1')
+        model1.SNAPSHOT_CONTENT_CLASS = None
+        with self.assertRaisesRegexp(
+            Exception, 'No snapshot content class defined.'):
+            model1.commit(feconf.SYSTEM_COMMITTER_ID, '', [])
+
+        model1 = TestVersionedModel(id='model_id1')
+        with self.assertRaisesRegexp(
+            Exception, 'Expected commit_cmds to be a list of dicts, received'):
+            model1.commit(feconf.SYSTEM_COMMITTER_ID, '', {})
+
+        model1 = TestVersionedModel(id='model_id1')
+        with self.assertRaisesRegexp(
+            Exception, 'Expected commit_cmds to be a list of dicts, received'):
+            model1.commit(feconf.SYSTEM_COMMITTER_ID, '', [[]])
+
+    def test_put_raises_not_implemented_error_for_versioned_models(self):
+        model1 = TestVersionedModel(id='model_id1')
+
+        with self.assertRaises(NotImplementedError):
+            model1.put()
+
+    def test_commit_with_invalid_change_list_raises_error(self):
+        model1 = TestVersionedModel(id='model_id1')
+
+        # Test for invalid commit command.
+        with self.assertRaisesRegexp(
+            Exception, 'Invalid commit_cmd:'):
+            model1.commit(
+                feconf.SYSTEM_COMMITTER_ID, '', [{'invalid_cmd': 'value'}])
+
+        # Test for invalid change list command.
+        with self.assertRaisesRegexp(
+            Exception, 'Invalid change list command:'):
+            model1.commit(feconf.SYSTEM_COMMITTER_ID, '', [{'cmd': 'AUTO'}])
+
+    def test_revert_with_not_allowed_revert_raises_error(self):
```

Comment:
Rename: `test_revert_raises_error_when_not_allowed`

In general, naming things in the following pattern should help you avoid us asking for changes:

`test_(what_happens)_(if|when|for|etc.)_(scenario)`

No need to follow it explicitly, just make sure your names answers the questions:
 - What happens?
 - Why/when does it happen?

+@seanlip for thoughts on anything else we should encourage/a better way to ensure good names?

Reply:
Done

Reply:
What @brianrodri says SGTM and is consistent with our existing guidelines for backend tests (see the [wiki page](https://github.com/oppia/oppia/wiki/Writing-backend-tests#guidelines-for-writing-good-tests)).